stingray/crosscorrelation.py

```diff
@@ -0,0 +1,190 @@
+from __future__ import division
+import numpy as np
+from scipy import signal
+
+from stingray import lightcurve
+from stingray.exceptions import StingrayError
+import stingray.utils as utils
+
+
+class CrossCorrelation(object):
+    def __init__(self, lc1=None, lc2=None):
+
+        """
+        Make a cross-correlation from a light curves.
+        You can also make an empty Crosscorrelation object to populate with your
+        own cross-correlation data.
+
+        Parameters
+        ----------
+        lc1: lightcurve.Lightcurve object, optional, default None
+            The first light curve data for correlation calculations.
+
+        lc2: lightcurve.Lightcurve object, optional, default None
+            The light curve data for the correlation calculations.
+
+        Attributes
+        ----------
+         corr: numpy.ndarray
+             An array of correlation data calculated from two lighcurves
+
+         time_lags: numpy.ndarray
+             An array of all possible time lags against which each point in corr is calculated 
+
+         dt: float
+             The time resolution of each lightcurve (used in time_lag calculations)
+
+         time_shift: float
+             Time lag that gives maximum value of correlation between two lightcurves.
+             There will be maximum correlation between lightcurves if one of the lightcurve is shifted by time_shift.
+                 
+         n: int
+             Number of points in self.corr(Length of cross-correlation data) 
+        """
+
+        ## Populate all attributes by None if user passes no lightcurve data
+        if lc1 is None and lc2 is None:
+            self.corr = None
+            self.time_shift = None
+            self.time_lags = None
+            self.dt = None
+            self.n = None
+            return
+        else:
+            self._make_corr(lc1, lc2)
+
+    def _make_corr(self, lc1, lc2):
+
+        """
+        Creates Crosscorrelation Object.
+
+        Parameters
+        ----------
+        lc1: lightcurve.Lightcurve object
+            The first light curve data.
+
+        lc2: lightcurve.Lightcurve object
+            The second light curve data.
+
+        Returns
+        ----------
+        None
+        """
+
+        if not isinstance(lc1, lightcurve.Lightcurve):
+            raise TypeError("lc1 must be a lightcurve.Lightcurve object")
+        if not isinstance(lc2, lightcurve.Lightcurve):
+            raise TypeError("lc2 must be a lightcurve.Lightcurve object")
+
+        # Sizes of both light curves are assumed to be equal for now
+        if lc1.n != lc2.n:
+            raise StingrayError('Both lightcurves should be of same length')
```

Comment:
I think time binning should be same for both light curves since correlation is calculated against time_lags with time resolution of `dt` (same as light curve binning). I think that requirement for light curves to have same size should not be there(If you have same thoughts, I can experiment with allowing different sizes of light curves and see how it behaves).